sky/cloud_stores.py

```diff
@@ -80,8 +80,7 @@ def make_sync_dir_command(self, source: str, destination: str) -> str:
 
     def make_sync_file_command(self, source: str, destination: str) -> str:
         """Downloads a file using AWS CLI."""
-        download_via_awscli = (f'mkdir -p {destination} &&'
-                               f'aws s3 cp {source} {destination}')
+        download_via_awscli = f'aws s3 cp {source} {destination}'
```

Comment:
What if `destination` contains several levels of dirs that haven't been created? Does `aws s3 cp` automatically mkdir?

Reply:
We added the `mkdir` for the destination in the `cloud_vm_ray_backend.py`, which works for both S3 and GCS. https://github.com/skypilot-org/skypilot/blob/15aa101f8deff32348a105cc216170bce76e4f37/sky/backends/cloud_vm_ray_backend.py#L2978-L2980 .

Actually, I am leaning towards having the `mkdir -p {os.path.dirname(destination)}` in the `make_sync_file_command` for both GCS and S3 instead of in the `cloud_vm_ray_backend.py`. What do you guys think @concretevitamin @romilbhardwaj ?

Reply:
Any reasons? `mkdir_for_wrapped_dst` is also used for non-cloud sources there, so it seems clearer if both cloud and non-cloud sources define some mkdir command in that func.

For this func here, maybe add a pointer to cloud_vm_ray_backend to be clearer?

Reply:
Got it. In that case, I removed the `mkdir -p` for the `make_sync_dir_command` function for s3 store as well.